cpp/src/arrow/scalar_test.cc

```diff
@@ -1077,21 +1077,38 @@ std::shared_ptr<DataType> MakeListType<FixedSizeListType>(
 
 template <typename ScalarType>
 void CheckListCast(const ScalarType& scalar, const std::shared_ptr<DataType>& to_type) {
-  EXPECT_OK_AND_ASSIGN(auto cast_scalar, scalar.CastTo(to_type));
-  ASSERT_OK(cast_scalar->ValidateFull());
-  ASSERT_EQ(*cast_scalar->type, *to_type);
+  EXPECT_OK_AND_ASSIGN(auto cast_scalar, Cast(scalar, to_type));
```

Comment:
How about using `cast_scalar_datum` for this variable? `cast_scalar.scalar()` is strange.

```suggestion
  EXPECT_OK_AND_ASSIGN(auto cast_scalar_datum, Cast(scalar, to_type));
  auto cast_scalar = cast_scalar_datum.scalar();
```

Reply:
I changed it.